### ISSUE TITLE
Button disabled states do not have enough contrast in 6.9 beta 2

### ISSUE THREAD

**Issue by t-hamano** [WordPress/gutenberg#72962]
- Originally reported at https://core.trac.wordpress.org/ticket/64191 by @mikemcalister
- Related to https://github.com/WordPress/gutenberg/pull/70674

> The disabled state of the button component in 6.9 beta 2 are quite hard to read. We either need to go darker on the background or darker on the text to create some more contrast. See the Publish/Save button in the editor toolbar.

**Comment by t-hamano** [comment]
For reference, this shows how the contrast ratio changes when a gray background is applied.

---

### Before: `1.9:1`

https://webaim.org/resources/contrastchecker/?fcolor=FFFFFF66&bcolor=007CBA

<img width="719" height="235" alt="Image" src="https://github.com/user-attachments/assets/5096b4c1-be37-49d4-ac77-c90e4c2f15f9" />

### After: `1.35:1`

<img width="723" height="237" alt="Image" src="https://github.com/user-attachments/assets/fe376948-0261-4704-b95b-1ba5305c22e8" />

https://webaim.org/resources/contrastchecker/?fcolor=FFFFFF&bcolor=DDDDDD

---

However, I'm not sure whether this is a bug or not. [This document](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html) states the following:

> ### Inactive User Interface Components
> 
> User Interface Components that are not available for user interaction (e.g., a disabled control in HTML) are **not required to meet contrast requirements**. An inactive user interface component is visible but not currently operable. An example would be a submit button at the bottom of a form that is visible but cannot be activated until all the required fields in the form are completed.

cc @WordPress/gutenberg-components

**Comment by joedolson** [comment]
It is permitted under WCAG for disabled components to have lower contrast, but this should still be considered judiciously. The allowance is because lower contrast is a helpful affordance that the component is disabled, but making it nearly impossible to see isn't really helpful. 

While decreasing the prominence of the disabled button makes sense, I think we should  try to make it a bit more readable than this. We should at least maintain the contrast we had before this change.

**Comment by t-hamano** [comment]
Thanks for the feedback! Let's try to fix this issue in the 6.9 release.

**Comment by aduth** [comment]
This is something I'm hoping to see resolved as part of the [theming package](https://github.com/WordPress/gutenberg/tree/trunk/packages/theme) color generation work.

For example, the default pairing of [background](https://github.com/WordPress/gutenberg/blob/9128c16ee0b94412bd55ba9b8216bac75009f97e/packages/theme/src/prebuilt/css/design-tokens.css#L13) and [foreground (text)](https://github.com/WordPress/gutenberg/blob/9128c16ee0b94412bd55ba9b8216bac75009f97e/packages/theme/src/prebuilt/css/design-tokens.css#L64) colors for a button would produce a [contrast of 2.62:1](https://webaim.org/resources/contrastchecker/?fcolor=898989&bcolor=DFDFDF).

**Comment by mikemcalister** [comment]
Thanks, @t-hamano for moving this over here! A 6.9 fix would be ideal because this button component is also used prolifically by 3rd party devs, so the implications are far reaching.

**Comment by aduth** [comment]
Looping in @jasmussen as the original author of #70674. With the planned theming pair, could we implement an interim fix by using a similar color combination with the existing color scales (specifically the text color). Say, [`$gray-600` on `$gray-300`](https://webaim.org/resources/contrastchecker/?fcolor=898989&bcolor=DFDFDF)?

**Comment by jasmussen** [comment]
I'm happy to be corrected by someone else, but the motivation for changing the color was specifically to reduce the prominence of a disabled button, and disabled buttons in my understanding do not need to meet contrast ratios:

> User Interface Components that are not available for user interaction (e.g., a disabled control in HTML) are not required to meet contrast requirements. An inactive user interface component is visible but not currently operable. An example would be a submit button at the bottom of a form that is visible but cannot be activated until all the required fields in the form are completed.

[W3C](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html).

To that end, before we choose new colors, I'd love to fully understand what's being asked for here.

**Comment by mikemcalister** [comment]
Thanks for the feedback, folks! 

To my eye, with these new colors, the disabled state is illegible (I thought it was broken at first glance). I understand that disabled states don't _need_ to adhere to the same contrast ratios, but they are still responsible for providing context and for that, they should be legible. 

Currently it's using gray 300 as the background. Bumping it to gray 400 or 600 could certainly help. But I think what could make it even more legible is using a darker text color. Here's what it loos like with keeping gray 300 as the background and gray 600 and 700 as the text. 

<img width="546" height="380" alt="Image" src="https://github.com/user-attachments/assets/ad6ad995-b6e9-4814-9ca7-8a1f6b54e8be" />

<img width="546" height="380" alt="Image" src="https://github.com/user-attachments/assets/b151cd06-5f77-4abb-bb45-8d2b4588777e" />

This has a nice balance of legibility and reduced prominence (especially over the very prominent blue disabled state). 

I don't have the full context of how these colors and patterns currently fit into the design system, so you'll have to take my implementation brainstorm with a grain of salt!

**Comment by joedolson** [comment]
I agree with @mikemcalister's take on what we need from the disabled state. Yes, it is allowed to be below contrast guidelines; but if it goes too far below that, it becomes illegible. 

I like the second version, with the slightly darker text. To me, that's clearly disabled, while still maintaining at least a reasonable degree of legibility. It still reduces prominence, but not as far.

To answer @jasmussen's request for clarification, we're just asking that the reduction in contrast is not this significant. And let's keep in mind that the WCAG 2.1 requirements are just a minimum baseline: they're what is *allowable*, not what is necessarily accessible.

**Comment by t-hamano** [comment]
Let's look at some variations.

---

- background: $gray-300 (#ddd), text: $gray-600 (#949494)
- Contrast Ratio: `2.23:1`

<details><summary>Screenshot</summary>

<img width="724" height="466" alt="Image" src="https://github.com/user-attachments/assets/68029415-d9cd-4c63-804a-fd6f747ccf9a" />

</details> 

---

- background: $gray-300 (#ddd), text: $gray-700 (#757575)
- Contrast Ratio: `3.39:1`

<details><summary>Screenshot</summary>

<img width="739" height="464" alt="Image" src="https://github.com/user-attachments/assets/7c85a90d-4a07-40f2-a815-cf9461abdbcc" />

</details>

---
 
- background: $gray-200 (#e0e0e0), text: $gray-600 (#949494)
- Contrast Ratio: `2.29:1`

<details><summary>Screenshot</summary>

<img width="731" height="470" alt="Image" src="https://github.com/user-attachments/assets/0d42022f-14e8-41ea-baf9-06ed4dbf2814" />

</details>

---
- background: $gray-200 (#e0e0e0), text: $gray-700 (#757575)
- Contrast Ratio: `3.49:1`

<details><summary>Screenshot</summary>

<img width="737" height="468" alt="Image" src="https://github.com/user-attachments/assets/094694b9-0875-45c8-acd3-b795997361d2" />

</details>

---

In all variations, the contrast ratio has been improved from the previous `1.9:1`.

The more we improve the contrast ratio, the more likely it is that users will mistakenly think the button is "not disabled." In particular, in the top four variations, where the text is `$gray-700`, the contrast ratio is significantly improved, but it may not be immediately obvious that the button is disabled.

Personally, I feel that the third variation, the one where the background is `$gray-200` and the text is `$gray-600`, is the most well-balanced.

**Comment by joedolson** [comment]
The ratio of 2.29:1 would be ok for me. I prefer the higher contrast options, but the 3rd variation would be a slightly better contrast than the previous blue disabled state, and considerably better than what we have right now, so I'd be totally willing to compromise on that.

**Comment by jasmussen** [comment]
Thank you all for the clarification here, much appreciate your time.

I'd love to narrow in on the colors as the next step, so that we can thread the needle between the original goal of reducing prominence, where a color token system might take this in the future, and finding something that does not look like an error.

For me, that's not dark gray text on a light gray background, mostly because that inversion and combination of colors is used anywhere else among the componentry. To that end, I'd first love broader input from @WordPress/gutenberg-design, and throw into the mix a temporary revert in case a clear winner doesn't emerge, so we can refine this in the 7.0 phase, perhaps _with_ new tokens. Let me know if this sounds reasonable, and then suggest this option:

<img width="272" height="132" alt="Image" src="https://github.com/user-attachments/assets/8dd638da-c2d5-4286-90f7-b9b52bd906cc" />

That gives us 2:1 contrast ratio between text and background. The motivation is this: it scales relatively easily, and without inventing too much new design language, to other componentry:

<img width="218" height="330" alt="Image" src="https://github.com/user-attachments/assets/18240618-4c41-410a-95ee-636011e01ad7" />

As noted, I invite feedback for this, but given color tokens are actively being developed to improve the whole system of color application, including the system for disabled colors, let me know what you think about a revert.

**Comment by t-hamano** [comment]
> given color tokens are actively being developed to improve the whole system of color application, including the system for disabled colors, let me know what you think about a revert.

For now, it might be best to revert the changes and reconsider them in accordance with [the new design system](https://github.com/WordPress/gutenberg/issues/71196). Personally, I vote +1 for reverting the changes.

**Comment by jasmussen** [comment]
If a revert resonates more broadly, I would appreciate help with that since I'm currently recording demo videos for the 6.9 marketing material (#71821).

**Comment by t-hamano** [comment]
This is a PR to revert to the previous one: https://github.com/WordPress/gutenberg/pull/73037

**Comment by mikemcalister** [comment]
That sounds great. Thanks for having an open mind and sharing some alternatives, @jasmussen. And thanks @t-hamano for the fix.
